src/gui/guiFormSpecMenu.cpp

```diff
@@ -2523,12 +2546,23 @@ void GUIFormSpecMenu::drawMenu()
 									AbsoluteRect.UpperLeftCorner.Y + absrec_size.Height + spec.pos.Y);
 			}
 
-			const video::SColor color(255,255,255,255);
-			const video::SColor colors[] = {color,color,color,color};
-			draw2DImageFilterScaled(driver, texture, rect,
-				core::rect<s32>(core::position2d<s32>(0,0),
-						core::dimension2di(texture->getOriginalSize())),
-				NULL/*&AbsoluteClippingRect*/, colors, true);
+			if (middle.getSize() == core::vector2di(0, 0)) {
```

Comment:
Why not
```suggestion
			if (middle.getArea() == 0) {
```
?

Reply:
I expect this could be more intensive. Maybe I should check to see if both sides are 0.

Reply:
[`getSize`](http://irrlicht.sourceforge.net/docu/rect_8h_source.html#l00231) calls `getWidth` and `getHeight` and creates a new dimension, [`getArea`](http://irrlicht.sourceforge.net/docu/rect_8h_source.html#l00097) also calls `getWidth` and `getHeight` and just multiplies both.